python/ray/dag/tests/experimental/test_torch_tensor_dag.py

```diff
@@ -1300,6 +1334,398 @@ def test_nccl_all_reduce_with_class_method_output_node(ray_start_regular):
+
+    num_workers = 2
+    workers = [actor_cls.remote() for _ in range(num_workers)]
+
+    with InputNode() as inp:
+        computes = [
+            worker.compute_with_tuple_args.bind(inp, i)
+            for i, worker in enumerate(workers)
+        ]
+        collectives = collective.reducescatter.bind(computes, ReduceOp.SUM)
+        recvs = [
+            worker.recv_tensor.bind(collective)
+            for worker, collective in zip(workers, collectives)
+        ]
+        dag = MultiOutputNode(recvs)
+
+    compiled_dag = dag.experimental_compile()
+
+    for i in range(3):
+        i += 1
+        shape = (num_workers * i, i)
+        dtype = torch.float16
+        ref = compiled_dag.execute(
+            [(shape, dtype, i + idx) for idx in range(num_workers)]
+        )
+        result = ray.get(ref)
+        reduced_val = sum(i + idx for idx in range(num_workers))
+        for tensor in result:
+            tensor = tensor.to("cpu")
+            expected_tensor_val = torch.ones((i, i), dtype=dtype) * reduced_val
+            assert torch.equal(tensor, expected_tensor_val)
+
+
+@pytest.mark.parametrize("ray_start_regular", [{"num_cpus": 4}], indirect=True)
+def test_torch_tensor_nccl_reduce_scatter_get_partial(ray_start_regular):
+    """
+    Test getting partial results from an reduce-scatter does not hang.
+    """
+    if not USE_GPU:
+        pytest.skip("NCCL tests require GPUs")
+
+    assert (
+        sum(node["Resources"].get("GPU", 0) for node in ray.nodes()) > 1
+    ), "This test requires at least 2 GPUs"
+
+    actor_cls = TorchTensorWorker.options(num_cpus=0, num_gpus=1)
+
+    num_workers = 2
+    workers = [actor_cls.remote() for _ in range(num_workers)]
+
+    with InputNode() as inp:
+        computes = [
+            worker.compute_with_tuple_args.bind(inp, i)
+            for i, worker in enumerate(workers)
+        ]
+        collectives = collective.reducescatter.bind(computes, ReduceOp.SUM)
+        tensor1 = workers[0].recv_tensor.bind(collectives[0])
+        tensor2 = workers[1].recv_tensor.bind(collectives[0])
+        dag = MultiOutputNode([tensor1, tensor2, collectives[1]])
+
+    compiled_dag = dag.experimental_compile()
+
+    for i in range(3):
+        i += 1
+        shape = (num_workers * i, i)
+        dtype = torch.float16
+        ref = compiled_dag.execute(
+            [(shape, dtype, i + idx) for idx in range(num_workers)]
+        )
+        result = ray.get(ref)
+        tensor1, tensor2, _ = result
+        reduced_val = sum(i + idx for idx in range(num_workers))
+        tensor1 = tensor1.to("cpu")
+        tensor2 = tensor2.to("cpu")
+        expected_tensor_val = torch.ones((i, i), dtype=dtype) * reduced_val
+        assert torch.equal(tensor1, expected_tensor_val)
+        assert torch.equal(tensor1, tensor2)
+
+
+@pytest.mark.parametrize("ray_start_regular", [{"num_cpus": 4}], indirect=True)
+def test_torch_tensor_nccl_reduce_scatter_different_shapes_among_participants(
```

Comment:
Is this test same as `all_reduce_wrong_shape`? If so, we should use the same names for both. `wrong_shape` is fine to keep.